### ISSUE TITLE
由淺入深的深度學習資源整理

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#12122]
由淺入深的深度學習資源整理<br>
<br>
https://ift.tt/OTQlMvK<br>
<br>
Lee Meng<br>
<br>

<div><div><div><div>
<div>
<div>
<div>
<div>
<div>
<div>
<blockquote>
<p>
不聞不若聞之，聞之不若見之，見之不若知之，知之不若行之，學至於行之而止矣。
<br><span>── 《荀子．儒效》<span>
</span></span></p>
</blockquote>
</div>
</div>
</div>
</div>
</div>
<div>
<p>
這段話翻成白話文就是「沒聽過比不上聽過；聽過比不上實際看過；看過則比不上實際了解；而了解又不如動手實踐。唯有身體力行才能真正地學到東西。」
</p>
<p>
這句古老的諺語向我們傳達了「實踐」的重要以及學習的幾個過程。
</p>
<p>
做為一門學問，<a href="https://zh.wikipedia.org/wiki/%E6%B7%B1%E5%BA%A6%E5%AD%A6%E4%B9%A0" target="_blank">深度學習</a>也是同樣道理。
僅說自己對深度學習有興趣或是有關注（聞、見），但卻沒有實際花時間去深入了解或實際應用（知、行）是無法真正學會深度學習的。
</p>
<p>
雖說如此，不了解深度學習能拿來做什麼的人或許還不少。
</p>
<p>
我嘗試將自己在學習過程中蒐集到的重要資源由淺入深地做些整理。
希望透過此文，能讓在各個學習階段的你都能從這裡獲得些什麼，並實際動手學習、探索發展快速的深度學習世界。
</p>
<p>
本文內容會持續被更新，你可以定期回來看看或是關注這個 <a href="https://github.com/leemengtaiwan/deep-learning-resources" target="_blank">Github Repo</a>。
</p>
</div>
<hr>
<p>這裡紀錄了我在學習<a href="https://leemeng.tw/deep-learning-for-everyone-understand-neural-net-and-linear-algebra.html">深度學習</a>時蒐集的一些線上資源。內容由淺入深，而且會不斷更新，希望能幫助你順利地開始學習：）</p>
<h2>本文章節</h2>
<h2></h2>
<p>遊玩空間</p>
<p>這節列舉了一些透過瀏覽器就能馬上開始遊玩 / 體驗深度學習的應用。作為這些應用的使用者，你可以先高層次、直觀地了解深度學習能做些什麼。之後有興趣再進一步了解背後原理。</p>
<p>這小節最適合：</p>
<hr>
<ul>
<li>一個利用<a href="https://zh.wikipedia.org/zh-hant/%E6%9C%BA%E5%99%A8%E5%AD%A6%E4%B9%A0">機器學習</a>來協助人們進行音樂以及藝術創作的開源專案</li>
<li>可以在網站上的 <a href="https://magenta.tensorflow.org/demos">Demo 頁面</a>嘗試各種由深度學習驅動的音樂 / 繪畫應用（如彈奏鋼琴、擊鼓）</li>
</ul>
<hr>
<ul>
<li>由 Google 推出的知名手寫塗鴉辨識，使用的神經網路架構有常見的<a href="https://zh.wikipedia.org/wiki/%E5%8D%B7%E7%A7%AF%E7%A5%9E%E7%BB%8F%E7%BD%91%E7%BB%9C">卷積神經網路 CNN </a>以及<a href="https://leemeng.tw/shortest-path-to-the-nlp-world-a-gentle-guide-of-natural-language-processing-and-deep-learning-for-everyone.html#%E6%9C%89%E8%A8%98%E6%86%B6%E7%9A%84%E5%BE%AA%E7%92%B0%E7%A5%9E%E7%B6%93%E7%B6%B2%E8%B7%AF_1">循環神經網路 RNN</a>
</li>
<li>該深度學習模型會不斷將最新的筆觸當作輸入來預測使用者想畫的物件。你會驚嘆於她精準且即時的判斷</li>
</ul>
<ul>
<li>利用電腦 / 手機上的相機來訓練能將影像對應到其他圖片、音訊的神經網路，饒富趣味</li>
<li>透過這例子，你將暸解機器學習的神奇之處以及其侷限所在</li>
</ul>
<hr>
<hr>
<hr>
<ul>
<li>一個偵測 / 生成神經假新聞（Neural Fake News）的研究，其網頁展示如何自動生成假新聞。</li>
</ul>
<hr>
<ul>
<li>Waifu 來自日文 ワイフ，指的是一些非常受到歡迎、且被不少玩家/觀眾視為妻子的動漫女性角色。<a href="https://twitter.com/SizigiStudios">Sizigi Studios</a> 團隊利用 GAN 隨機初始 16 名虛擬動漫角色，讓使用者可以進一步依照喜愛來創造專屬於自己的 Waifu。</li>
<li>Waifu Vending Machine 產生的 Waifu 品質很高，使用者可以下載並分享自己創造的 Waifu，也可以選擇購買印製該 Waifu 的海報與抱枕。</li>
</ul>
<hr>
<ul>
<li>Anomagram 是一個以 Tensorflow.js 實作，可以建立、訓練並測試能夠用來做異常檢測的 Autoencoder。</li>
</ul>
<h2></h2>
<p>線上課程</p>
<p>看完<a href="https://leemeng.tw/deep-learning-resources.html#playground">遊玩空間</a>的大量實際應用，相信你已經迫不及待地想要開始學習強大的深度學習技術了。</p>
<p>這節列舉了一些有用的線上課程以及學習教材，幫助你掌握深度學習的基本知識（沒有特別註明的話皆為免費存取）。</p>
<p>另外值得一提的是，大部分課程都要求一定程度的 <a href="https://www.python.org/">Python</a> 程式能力。</p>
<hr>
<ul>
<li>7 週課程，一週約需安排 10 小時上課。該課程由<a href="https://www.kaggle.com/jhoward">傑里米·霍華德</a>來講解深度學習，其在知名數據建模和數據分析競賽平台 <a href="https://www.kaggle.com/">Kaggle</a> 維持兩年的世界第一</li>
</ul>
<hr>
<ul>
<li>芬蘭最高學府<a href="https://zh.wikipedia.org/wiki/%E8%B5%AB%E5%B0%94%E8%BE%9B%E5%9F%BA%E5%A4%A7%E5%AD%A6">赫爾辛基大學</a>推出的 AI 課程。此課程目的在於讓所有人都能了解 AI，不需要任何程式經驗。這堂課非常適合完全沒有接觸過深度學習或是相關領域的人</li>
<li>課程分 6 個部分，包含「何謂 AI ？」、「真實世界的 AI」、「機器學習」以及「神經網路」等章節</li>
</ul>
<ul>
<li>麻省理工學院推出的深度學習課程，內容包含深度學習基礎、深度強化學習以及自動駕駛相關知識。<a href="https://github.com/lexfridman/mit-deep-learning">Github Repo</a> 包含了多個教學筆記本，值得參考。</li>
<li>上圖是 <a href="https://selfdrivingcars.mit.edu/deeptraffic/">DeepTraffic</a>，由 MIT 的研究科學家 <a href="https://lexfridman.com/">Lex Fridman</a> 推出的一個深度強化學習競賽。此競賽目標是建立一個可以在高速公路上駕駛汽車的神經網路。你可以在<a href="https://selfdrivingcars.mit.edu/deeptraffic/">這裡</a>看到線上 Demo 以及詳細說明。</li>
</ul>
<hr>
<ul>
<li>麻省理工學院推出的另一堂基礎深度學習課程，介紹深度學習以及其應用。內容涵蓋機器翻譯、圖像辨識以及更多其他應用。此課程使用 Python 以及 TensorFlow 來實作作業，並預期學生具備基礎的微積分（梯度下降、鏈鎖律）以及線性代數（矩陣相乘）。</li>
</ul>
<hr>
<hr>
<ul>
<li>MFML 是一門對 Google 內部開授的 ML/AI 訓練課程，由其 Chief Decision Scientist <a href="https://kozyrkov.medium.com/">Cassie Kozyrkov</a> 主講。此課程於 <a href="https://decision.substack.com/p/full-course-making-friends-with-ml">2021/11/25 對外開放</a>。</li>
<li>此課程提供從構思 ML 專案到 launch AI 產品的步驟與大局觀，十分適合剛接觸相關領域的產品經理與企業決策者入門。</li>
</ul>
<h2></h2>
<p>實用工具</p>
<p>這節列出一些在你的深度學習路上可以幫得上些忙的工具。</p>
<ul>
<li>由 Google 提供的雲端 <a href="https://jupyter.org/">Jupyter</a> 筆記本環境，讓你只要用瀏覽器就能馬上開始訓練深度學習模型。你甚至還可以使用一個免費的 <a href="https://www.nvidia.com/en-gb/data-center/tesla-k80/">Tesla K80</a> GPU 或 <a href="https://colab.research.google.com/notebooks/tpu.ipynb">TPU</a> 來加速訓練自己的模型</li>
<li>該計算環境也能與自己的 <a href="https://colab.research.google.com/notebooks/io.ipynb">Google Drive</a> 做連結，讓運算雲端化的同時將筆記本 / 模型結果都同步到自己的筆電上</li>
</ul>
<ul>
<li>TensorBoard 是一個視覺化工具，方便我們了解、除錯並最佳化自己訓練的深度學習模型</li>
<li>除了 TensorFlow 以外，其他基於 Python 的機器學習框架大多也可以透過 <a href="https://github.com/lanpa/tensorboardX">tensorboardX</a> 來使用 TensorBoard</li>
</ul>
<hr>
<ul>
<li>Lucid 是一個嘗試讓神經網路變得更容易解釋的開源專案，裡頭包含了很多視覺化神經網路的筆記本</li>
<li>你可以直接在 Colab 上執行這些筆記本並了解如何視覺化神經網路</li>
</ul>
<hr>
<ul>
<li>將機器學習的學術論文、程式碼實作以及 SOTA 的評價排行榜全部整理匯總在一起的網站，非常適合想要持續追蹤學術及業界最新研究趨勢的人</li>
<li>在這邊可以瀏覽包含電腦視覺、自然語言處理等各大領域在不同任務上表現最好的論文、實作以及資料集</li>
</ul>
<ul>
<li>一個與 <a href="https://leemeng.tw/deep-learning-resources.html#tensorboard">TensorBoard</a> 以及 Jupyter Notebook 整合的探索工具，讓使用者不需寫程式碼就能輕鬆觀察機器學習模型的內部運作以及嘗試各種 What-if 問題（如果 ~ 會怎麼樣？）</li>
<li>基本上就是用來觀察<strong>已訓練</strong>的模型在測試資料集上的表現。利用此工具，使用者可以了解（不僅限於）以下的問題：模型在各類別數據上的表現有無差距？模型是否存在偏見？應該如何調整 Native / Positive False 的比例？</li>
<li>此工具的一大亮點在於讓非專業領域人士也能探索、理解 ML 模型表現。且只要給定模型與資料集, 就不需要每次為了 What-if 問題就寫用過即丟的程式碼</li>
</ul>
<hr>
<ul>
<li>ML Visuals 是一個社群開源項目，提供超過 100 個常見的機器學習概念 / 深度學習架構圖，可讓任何人在學術論文或是文章直接使用這些圖表。</li>
<li>所有圖表都可以直接從 <a href="https://docs.google.com/presentation/d/11mR1nkIR9fbHegFkcFq8z9oDQ5sjv8E3JJp1LfLGKuk/edit?usp=sharing">Google slide</a> 上觀看並使用。建議前往 <a href="https://github.com/dair-ai/ml-visuals">Github repo</a> 查看最新版本。</li>
</ul>
<h2></h2>
<p>其他教材</p>
<p>除了<a href="https://leemeng.tw/deep-learning-resources.html#courses">線上課程</a>以外，網路上還有無數的學習資源。</p>
<p>這邊列出一些推薦的深度學習教材，大多數皆以數據科學家常用的 <a href="https://jupyter.org/">Jupyter</a> 筆記本的方式呈現。</p>
<p>你可以將感興趣的筆記本導入<a href="https://leemeng.tw/deep-learning-resources.html#tools">實用工具</a>裡提到的 <a href="https://colab.research.google.com/notebooks/welcome.ipynb">Colaboratory（Colab）</a>，馬上開始學習。</p>
<ul>
<li>讓你可以一覽 Colab 上超過 100 個跟機器學習相關的筆記本，並以此為基礎建立各種深度學習應用</li>
<li>熱門筆記本包含<a href="https://research.google.com/seedbank/seed/5695159920492544">神經機器翻譯</a>、<a href="https://research.google.com/seedbank/seed/5681034041491456">音樂生成</a>以及 <a href="https://research.google.com/seedbank/seed/5631986051842048">DeepDream</a>
</li>
<li>因為是 Google 服務，筆記本大多使用 TensorFlow 與 Keras 來實現模型</li>
</ul>
<hr>
<ul>
<li>在 Github 上超過 1 萬星的 Repo。除了深度學習，也有介紹 <a href="https://colab.research.google.com/github/practicalAI/practicalAI/blob/master/notebooks/basic_ml/01_Python.ipynb">Python 基礎</a>及 <a href="https://colab.research.google.com/github/practicalAI/practicalAI/blob/master/notebooks/basic_ml/03_Pandas.ipynb">Pandas</a> 的使用方式</li>
<li>使用 <a href="https://pytorch.org/">PyTorch</a> 框架來實現深度學習模型，且所有內容都是 Jupyter 筆記本，可以讓你在 Colab 或本地端執行</li>
</ul>
<hr>
<ul>
<li>清楚地展示了如<a href="https://demo.allennlp.org/machine-comprehension">機器理解</a>、<a href="https://demo.allennlp.org/named-entity-recognition">命名實體識別</a>等多個自然語言處理任務的情境。每個任務的情境包含了任務所需要的輸入、SOTA 模型的預測結果以及模型內部的注意力機制，對理解一個 NLP 任務的實際應用情境有很大幫助</li>
<li>
<a href="https://allennlp.org/">AllenNLP</a> 是一個由 <a href="https://allenai.org/">AI2</a> 以 <a href="https://pytorch.org/">PyTorch</a> 實現的自然語言處理函式庫</li>
</ul>
<ul>
<li>前 YouTube 影片分類 PM <a href="https://twitter.com/aureliengeron">Aurélien Geron</a> 教你如何透過 Scikit-Learn、Keras 以及 TensorFlow 2 來進行機器學習以及深度學習任務與應用的筆記本彙整。</li>
<li>第二版專注在 TensorFlow 2，其 Github repo 已有超過 6 千顆星，<a href="https://github.com/ageron/handson-ml">第一版</a>則有高達 2 萬星。</li>
</ul>
<h2></h2>
<p>優質文章</p>
<p>這邊列舉了一些幫助我釐清重要概念的部落格以及網站，希望能加速你探索這個深度學習世界。</p>
<p>只要 Google 一下就能發現這些部落格裡頭很多文章都有中文翻譯。但為了尊重原作者，在這邊都列出原文連結。</p>
<h2></h2>
<p>經典論文</p>
<p>這邊依發表時間列出深度學習領域的經典 / 重要論文。</p>
<p>為了幫助你快速掌握論文內容以及歷年的研究趨勢，每篇論文下會有非常簡短的介紹（WIP）。</p>
<p>但我們推薦有興趣的人自行閱讀論文以深入了解。</p>
<h3>自然語言處理 Natural Language Processing (NLP)</h3>
<h3>電腦視覺 Computer Vision (CV)</h3>
<h4>類神經網路架構 Neural Network Architecture</h4>
<h4>資料集 Dataset</h4>
<h4>物體偵測與切割 Object Detection and Segmentation</h4>
<h4>生成模型 Generative Models</h4>
<h2></h2>
<p>其他整理</p>
<p>這邊列出其他優質的資源整理網站 / Github Repo，供你繼續探索深度學習。</p>
<ul>
<li>整理了不少深度學習資源，但最值得參考的是數據集以及論文的分類整理。</li>
</ul>
<h2>待辦事項</h2>
<p>還有不少內容正在整理，以下是目前我們打算增加的一些項目：</p>
<ul>
<li>深度學習中英術語對照表</li>
<li>值得追蹤的業界 / 學界影響人物清單</li>
<li>無圖的資源列表版本</li>
<li>一些 Jupyter Notebook 範例</li>
</ul>
<p>而我們也會持續將新資源加入如<a href="https://leemeng.tw/deep-learning-resources.html#tools">實用工具</a>、<a href="https://leemeng.tw/deep-learning-resources.html#blogs">優質文章</a>等列表裡頭。</p>
<h2>如何貢獻</h2>
<p>非常歡迎你一起加入改善這個 Repo，讓更多人有方向地學習 Deep Learning：）</p>
<p>如果你有</p>
<ul>
<li>其他值得推薦的深度學習資源</li>
<li>針對此 Repo 內容的改善建議</li>
<li>其他任何你想得到的東西</li>
</ul>
<p>都歡迎你<a href="https://github.com/leemengtaiwan/deep-learning-resources/issues/new">提出新的 Issue</a> 來讓我們知道。</p>
<p>如果是想增加新資源的話，只附上連結也是沒有問題的，謝謝！
</p>
<div>
<p>
最後，如果你覺得本文實用，還請幫我分享此文並給 <a href="https://github.com/leemengtaiwan/deep-learning-resources" target="_blank">Github Repo</a> 一個小星星。
這樣可以讓更多人注意到這些寶貴資源的存在並開始有方向的學習，謝謝！
</p>
<p>
有再多資源，沒有親自動手做都是無法真正地學到東西的。因此，最後的最後讓我再次強調主動學習的重要：
</p>
</div>
<div>
<div>
<div>
<div>
<div>
<blockquote>
<p>
告訴我資訊，我只會忘記；教導我知識，我會記得；讓我實際參與，我將能真正地學到東西。
<br><span>── 班傑明·富蘭克林<span>
</span></span></p>
</blockquote>
</div>
</div>
</div>
</div>
</div>
<p>
<span>Post Tags</span>
<span>
<a rel="tag" href="https://leemeng.tw/tag/shen-du-xue-xi.html">深度學習</a>
<a rel="tag" href="https://leemeng.tw/tag/python.html">Python</a>
<a rel="tag" href="https://leemeng.tw/tag/keras.html">Keras</a>
<a rel="tag" href="https://leemeng.tw/tag/tensorflow.html">TensorFlow</a>
</span>
</p>
<div>
<div>
<p>
</p>
<div>
<p>
跟資料科學相關的最新文章直接送到家。
只要加入訂閱名單，當新文章出爐時，
你將能馬上收到通知 </p>
</div>
</div>
<p>
</p>
</div>
</div>
</div></div></div></div><br>
<br>
<br>
<br>
<br>
<br>
via LeeMeng - All About DataScience <br>
<br>
April 21, 2025 at 01:41PM
